### PR TITLE
[LTC] Further refactor bridge's device management

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This commit removes all unused device management bridge utils and consolidates
some existing ones.

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

### CODE REVIEW

File: lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp [hunk 1]
```diff
@@ -88,6 +88,16 @@ at::Tensor CreateLtcTensor(const at::Tensor& tensor,
   return tensor;
 }
 
+c10::optional<torch::lazy::BackendDevice> GetLtcDevice(const c10::optional<c10::Device>& device) {
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp:91]: Is this still needed?

File: lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp [hunk 2]
```diff
@@ -376,7 +386,7 @@ at::Tensor LazyNativeFunctions::empty(
                                   .pinned_memory(pin_memory)
                                   .dtype(dtype);
   auto x_result = at::empty(size, options, memory_format);
-  return CreateLtcTensor(x_result, bridge::GetLtcDevice(device));
+  return CreateLtcTensor(x_result, GetLtcDevice(device));
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp:389]: I guess it's still being called, but, i'm not sure what the reason is

**Review by wconstab — APPROVED** [review]

File: lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp (hunk 1 above)
Comment (alanwaketan) [reply lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp:91]: I moved it to aten_ltc_ts_type.cpp.

File: lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp (hunk 2 above)
Comment (alanwaketan) [reply lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/aten_ltc_ts_type.cpp:389]: Yea, it's more like a helper to do some checks before the actual conversion.

**Comment by alanwaketan** [comment]
Thanks for approving the pull request, Will.

**Pull request merged**